### PR TITLE
Rename MLLP.DefaultDispatcher to MLLP.EchoDispatcher

### LINKED ISSUE #17: Rename DefaultDispatcher to EchoDispatcher
In EchoDispatcher, we should remove `Logger.error` message. We should echo the message as info and returns positive ACK. It may require some changes in README as well.

_Originally posted by @vikas15bhardwaj in https://github.com/HCA-Healthcare/elixir-mllp/pull/16#discussion_r755466773_

### PR DESCRIPTION
**Pull request by starbelly**
- changed MLLP.DefaultDispatcher to MLLP.EchoDispatcher
 - updated references where required
 - updateed README.md
 
 Closes #17

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ Next, start an MLLP.Sender process and store its PID.
 
 Alternatively, you could start a Sender using a DNS name rather than an IP address.
 ```
-{:ok, s1} = MLLP.Sender.start_link("localhost", 4090)
+{:ok, s1} = MLLP.Sender.start_link("localhost", port: 4090, MLLP.EchoDispatcher)
```
Comment (vikas15bhardwaj) [inline README.md]: We don't need dispatched on `Sender.start_link`

Comment (vikas15bhardwaj) [reply README.md]: Also the port stays as parameter on Sender

File: lib/mllp/echo_dispatcher.ex [hunk 2]
```diff
@@ -1,7 +1,8 @@
-defmodule MLLP.DefaultDispatcher do
+defmodule MLLP.EchoDispatcher do
```
Comment (bryanhunter) [inline lib/mllp/echo_dispatcher.ex:1]: By renaming here you will also need to update the docs (receiver.ex line 51) to use the new name.

As is `iex(1)> h MLLP.Receiver.start` tells user to do it wrong

File: README.md (hunk 1 above)
Comment (starbelly) [reply README.md]: Good catch.

File: lib/mllp/echo_dispatcher.ex (hunk 2 above)
Comment (starbelly) [reply lib/mllp/echo_dispatcher.ex:1]: Good catch.

**Comment by vikas15bhardwaj** [comment]
Couple of places, I still see DefaultDispatcher in comments. README line 50, and 104

File: README.md [hunk 3]
```diff
@@ -55,7 +55,11 @@ Now, we will stop the Receiver.
  MLLP.Receiver.stop(4090)
 ```
 
-### Custom message dispatcher
+### Writing a message dispatcher
+
+MLLP does not ship with a default message dispatcher as the cases can vary signficantly from domain to domain. Instead,
```
Comment (bryanhunter) [inline README.md]:
```suggestion
MLLP does not ship with a default message dispatcher as the cases can vary significantly from domain to domain. Instead,
```

File: README.md [hunk 4]
```diff
@@ -47,15 +47,19 @@ You will see log info like...
  }}
 ```
 
-The Logger debug part tells us that the Receiver received the HL7 message. The Logger error part says we haven't set a Dispatcher and the message will not be routed anywhere other than to the console log. Finally, the return value (`{:error,  :application_reject, ...}`) is a NACK. The DefaultDispatcher will not reply with an `:application_accept`.
+The Logger debug part tells us that the Receiver received the HL7 message. The Logger error part says we haven't set a Dispatcher and the message will not be routed anywhere other than to the console log. Finally, the return value (`{:error,  :application_reject, ...}`) is a NACK. The EchoDispatcher will not reply with an `:application_accept`.
```
Comment (bryanhunter) [inline README.md]:
```suggestion
The Logger `[debug]` part tells us that the Receiver received the HL7 message. The Logger `[info]` part explains that the EchoDispatcher does not route the message anywhere (just "logs and discards" messages). Finally, the return value (`{:error,  :application_reject, ...}`) is a NACK. The EchoDispatcher will not reply with an `:application_accept`.
```

**Review by vikas15bhardwaj — APPROVED** [review]

**Pull request merged**